### PR TITLE
Enhancement #672: allow users to change overview exit transition animation

### LINKED ISSUE #672: Disable Overview transition animation on window selection
**Is your feature request related to a problem? Please describe.**
Sometimes selecting a window from the overview is much faster than navigating using the keyboard. But each time I select a window from the overview when using PaperWM, the overview translation animation runs which gets disorienting after multiple uses.

**Describe the solution you'd like**
Instead of animating the translation to the window selected from the overview, there can be an option to just fade in the window.

**Describe alternatives you've considered**
- Speeding up the animations for the GNOME shell
- Disabling animations
- Speeding up animations in PaperWM

**Additional context**
https://github.com/paperwm/PaperWM/assets/26873863/9778a130-a13d-4d54-b51b-ad9fb07d22f1

### PR DESCRIPTION
**Pull request by jtaala**
This PR fixes #672.

Namely, it now allows users to select from PaperWM settings the animation used to ensure the selected window (after exiting Gnome Overview) is brought into view:

image

### CODE REVIEW

**Comment by Lythenas** [comment]
Is the fade supposed to show the workspace background and then fade in the window? That seems a bit strange to me and not really better than no animation.

**Comment by jtaala** [comment]
> Is the fade supposed to show the workspace background and then fade in the window? That seems a bit strange to me and not really better than no animation.

No, it's pretty much like no animation but a little softer (it just fades window out on exiting overview, and fades it back in).  This was the preferred one by the original submitter:

https://github.com/paperwm/PaperWM/issues/672#issuecomment-1801508556

**Review by Lythenas — APPROVED** [review]
Then I'm fine with this.

File: Settings.ui [hunk 1]
```diff
@@ -1114,6 +1114,48 @@
...
                       </object>
                     </child>
+                    <child>
+                      <object class="GtkListBoxRow">
+                        <property name="activatable">False</property>
+                        <property name="focusable">False</property>
+                        <child>
+                          <object class="GtkGrid">
+                            <property name="focusable">False</property>
+                            <property name="tooltip_text" translatable="yes">Sets the animation when exiting Gnome overview and ensuring selected window is in view</property>
```
Comment (Thesola10) [inline Settings.ui]:
```suggestion
                            <property name="tooltip_text" translatable="yes">Sets the animation when exiting GNOME overview and ensuring selected window is in view</property>
```

File: schemas/org.gnome.shell.extensions.paperwm.gschema.xml [hunk 2]
```diff
@@ -449,6 +449,11 @@
         <summary>Duration of animations in seconds</summary>
     </key>
 
+    <key type="i" name="overview-ensure-viewport-animation">
+        <default>1</default>
+        <summary>Sets the animation when exiting Gnome overview and ensuring selected window is in view. 0:NONE, 1:TRANSLATE, 2:FADE</summary>
```
Comment (Thesola10) [inline schemas/org.gnome.shell.extensions.paperwm.gschema.xml]:
```suggestion
        <summary>Sets the animation when exiting GNOME overview and ensuring selected window is in view. 0:NONE, 1:TRANSLATE, 2:FADE</summary>
```

**Review by Thesola10 — APPROVED** [review]
oh, this only applies if tiled windows are shown in Overview

could this be used to address #652 as well?

**Comment by jtaala** [comment]
> oh, this only applies if tiled windows are shown in Overview
> 
> could this be used to address #652 as well?

No, unfortunately that would be a separate issue.  These changes don't effect the actual overview animations - they happen after the overview animation has finished (when bring the selected window into tiled view).

**Comment by jtaala** [comment]
> > could this be used to address #652 as well?

The issue with #652 is that it's the default gnome behaviour (to just dissappear windows that normally wouldn't display in overview).  E.g. here's with PaperWM disabled (note I've disabled attaching modal dialogs to windows):

https://github.com/paperwm/PaperWM/assets/30424662/b8d01bf1-271f-417e-aa1c-827272804f71

So to change #652 we'd need to override or modify gnome overview animation (entry and exits).

**Review by Thesola10 — APPROVED** [review]

**Pull request merged**